src/main/java/dev/pgm/community/polls/feature/PollFeature.java

```diff
@@ -0,0 +1,548 @@
+package dev.pgm.community.polls.feature;
+
+import static net.kyori.adventure.text.Component.text;
+
+import com.google.common.collect.Lists;
+import dev.pgm.community.Community;
+import dev.pgm.community.feature.FeatureBase;
+import dev.pgm.community.polls.Poll;
+import dev.pgm.community.polls.PollBuilder;
+import dev.pgm.community.polls.PollConfig;
+import dev.pgm.community.polls.PollEditAlerter;
+import dev.pgm.community.polls.ending.EndAction;
+import dev.pgm.community.polls.ending.types.NullEndAction;
+import dev.pgm.community.polls.types.TimedPoll;
+import dev.pgm.community.utils.BroadcastUtils;
+import dev.pgm.community.utils.CenterUtils;
+import dev.pgm.community.utils.CommandAudience;
+import dev.pgm.community.utils.MessageUtils;
+import dev.pgm.community.utils.Sounds;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.logging.Logger;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.TextComponent;
+import net.kyori.adventure.text.event.ClickEvent;
+import net.kyori.adventure.text.event.HoverEvent;
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.kyori.adventure.text.format.TextColor;
+import net.kyori.adventure.text.format.TextDecoration;
+import org.bukkit.Bukkit;
+import org.bukkit.configuration.Configuration;
+import org.bukkit.entity.Player;
+import tc.oc.pgm.util.Audience;
+import tc.oc.pgm.util.text.TemporalComponent;
+import tc.oc.pgm.util.text.TextFormatter;
+
+public class PollFeature extends FeatureBase implements PollEditAlerter {
+
+  private PollBuilder builder;
+  private Poll poll;
+
+  private boolean delayedStart;
+  private Integer delayedTaskID;
+
+  public PollFeature(Configuration config, Logger logger) {
+    super(new PollConfig(config), logger, "Polls");
+
+    if (getConfig().isEnabled()) {
+      enable();
+      Bukkit.getScheduler().scheduleSyncRepeatingTask(Community.get(), this::task, 0L, 20L);
+    }
+  }
+
+  private void task() {
+    if (!isRunning()) return;
+
+    if (poll instanceof TimedPoll) {
+      TimedPoll timedPoll = (TimedPoll) poll;
+      Duration timeLeft = timedPoll.getTimeLeft();
+
+      if (shouldShowAlert(timeLeft)) {
```

Comment:
You should consider having a different alert frequency for players who have already voted vs players who haven't voted.